test/base/src/androidTest/kotlin/io/realm/kotlin/test/shared/MutableRealmIntTests.kt

```diff
@@ -0,0 +1,290 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.test.shared
+
+import io.realm.kotlin.Realm
+import io.realm.kotlin.RealmConfiguration
+import io.realm.kotlin.entities.Sample
+import io.realm.kotlin.internal.RealmObjectReference
+import io.realm.kotlin.test.assertFailsWithMessage
+import io.realm.kotlin.test.platform.PlatformUtils
+import io.realm.kotlin.types.MutableRealmInt
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertNotEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNotSame
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+class MutableRealmIntTests {
+
+    private lateinit var tmpDir: String
+    private lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+        val configuration = RealmConfiguration.Builder(setOf(Sample::class))
+            .directory(tmpDir)
+            .build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (this::realm.isInitialized && !realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun unmanaged_basic() {
+        val sample1 = Sample().apply { nullableMutableRealmIntField = null }
+        val sample2 = Sample().apply { nullableMutableRealmIntField = null }
+        basicTest(sample1, sample2)
+    }
+
+    @Test
+    fun unmanaged_equality() {
+        val sample1 = Sample()
+        val sample2 = Sample()
+        equalityTest(sample1, sample2)
+    }
+
+    @Test
+    fun unmanaged_nullability() {
+        nullabilityTest(Sample())
+    }
+
+    @Test
+    fun unmanaged_compareTo() {
+        val r1 = MutableRealmInt.of(0)
+        val r2 = MutableRealmInt.of(Long.MAX_VALUE)
+        assertEquals(-1, r1.compareTo(r2))
+        r2.decrement(Long.MAX_VALUE)
+        assertEquals(0, r1.compareTo(r2))
+        r2.decrement(Long.MAX_VALUE)
+        assertEquals(1, r1.compareTo(r2))
+    }
+
+    @Test
+    fun unmanaged_shareValueAcrossInstances() {
+        val counter = MutableRealmInt.of(42)
+        val foo = Sample()
+        val bar = Sample()
+
+        foo.mutableRealmIntField = counter
+        bar.mutableRealmIntField = foo.mutableRealmIntField
+        bar.mutableRealmIntField.increment(1)
+        val fooValue = foo.mutableRealmIntField.get()
+        val barValue = bar.mutableRealmIntField.get()
+
+        assertEquals(fooValue, barValue)
+    }
+
+    @Test
+    fun managed_basic() {
```

Comment:
Same as unmanaged, these tests do not seem relevant for testing MutableRealmInt?